### PR TITLE
added Snake Game

### LINKED ISSUE #638: Kitchen Sink: Build A Snake Game
### Describe the feature

I would like to develop a Snake game and incorporate it into this repository. The game will display the player's score as well as the highest score achieved in previous games.
could you kindly assign this issue to me under the hactoberfest label.

### Additional information

- [x] Would you be willing to help implement this feature?

### PR DESCRIPTION
**Pull request by sr2005roy**
**Please describe the changes this PR makes and why it should be merged:**

Kitchen Sink: Added A Snake Game
resolves #638
closed: [Kitchen Sink: Build A Snake Game](https://github.com/aidenybai/million/issues/638)
images:
image
image

**Status**

- [x] Code changes have been tested against prettier, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**

- [x] This PR changes the codebase
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
  - [x] This PR changes the internal workings with no modifications to the external API (bug fixes, performance improvements)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by sr2005roy** [comment]
@tobySolutions can you please check the PR under hactoberfest-accepted label

**Comment by tobySolutions** [comment]
> @tobySolutions can you please check the PR under hactoberfest-accepted label

Reviewing this now. Thanks @sr2005roy.

File: packages/kitchen-sink/src/examples/snakeGame.jsx [hunk 1]
```diff
@@ -0,0 +1,187 @@
+import { block } from 'million/react';
+import { useEffect, useRef, useState } from "react";
+
+const Snake = ({ snake }) => {
```
Comment (tobySolutions) [inline packages/kitchen-sink/src/examples/snakeGame.jsx:4]: You could make this be a block

File: packages/kitchen-sink/src/examples/snakeGame.jsx [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+            top: `${box.y}%`,
+            zIndex: 1,
+          }}
+        />
+      ))}
+    </div>
+  );
+}
+
+const Food = ({ position }) => {
```
Comment (tobySolutions) [inline packages/kitchen-sink/src/examples/snakeGame.jsx:25]: This guy too!

File: packages/kitchen-sink/src/style.css [hunk 3]
```diff
@@ -599,8 +599,8 @@ input[type='range'] {
   opacity: 1;
 }
 
-.password_generator_customizable_options {
-}
```
Comment (tobySolutions) [inline packages/kitchen-sink/src/style.css:603]: Did you comment out code here by chance?

**Review by tobySolutions — APPROVED** [review]
In summary, your changes LGTM!!

Could you maybe check out the responsiveness on mobile maybe?

Screenshot 2023-10-14 at 19 40 29

**Comment by tobySolutions** [comment]
Any updates on this @sr2005roy?

**Comment by sr2005roy** [comment]
I have made all the changes

**Comment by sr2005roy** [comment]
Now kindly check @tobySolutions

**Comment by akshayw1** [comment]
@sr2005roy Need a help in order to make PR in kitchen sink? Could you pls explain in steps

**Comment by tobySolutions** [comment]
> Now kindly check @tobySolutions

Awesome! I'm gonna check this out now.

**Review by tobySolutions — APPROVED** [review]
LGTM! Awesome!

**Pull request merged**